transaction/std/src/transaction_builder.rs

```diff
@@ -108,12 +111,16 @@ impl TransactionBuilder {
         Ok((tx_out, confirmation))
     }
 
-    /// Sets the tombstone block.
+    /// Sets the tombstone block, clamping to smallest pubkey expiry value.
     ///
     /// # Arguments
     /// * `tombstone_block` - Tombstone block number.
-    pub fn set_tombstone_block(&mut self, tombstone_block: u64) {
-        self.tombstone_block = tombstone_block;
+    pub fn set_tombstone_block(&mut self, tombstone_block: u64) -> u64 {
+        self.tombstone_block = core::cmp::min(
+            tombstone_block,
+            self.fog_pubkeys.smallest_pubkey_expiry_value(),
+        );
+        self.tombstone_block
```

Comment:
another option here might be to return an error if it exceeds smallest_pubkey_expiry_value

but the caller can also easily test if clamping occurred

Reply:
Ditto about  MAX_TOMBSTONE_BLOCKS